### PR TITLE
feat(RDS): add RDS PG plugin resource

### PR DESCRIPTION
**Pull request by liwanting0517**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add RDS PG plugin resource
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rds' TESTARGS='-run TestAccPgPlugin_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds -v -run TestAccPgPlugin_basic -timeout 360m -parallel 4
=== RUN   TestAccPgPlugin_basic                                           TF_ACC=1 go test ./huaweicloud/services/acceptance/rds -v -run TestAccPgPlugin_basic -timeout 360m -pa
rallel 4
=== RUN   TestAccPgPlugin_basic
=== PAUSE TestAccPgPlugin_basic
=== CONT  TestAccPgPlugin_basic
--- PASS: TestAccPgPlugin_basic (497.24s)
PASS
```

### CODE REVIEW

File: docs/resources/rds_pg_plugin.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 30 minutes.
+* `delete` - Default is 30 minutes.
+
+## Import
+
+The RDS for PostgreSQL plugin can be imported using the `instance_id`, `database_name` and `name` separated by a slash, e.g.:
```
Comment (houpeng80) [inline docs/resources/rds_pg_plugin.md]:
```suggestion
The RDS for PostgreSQL plugin can be imported using the `instance_id`, `database_name` and `name` separated by slashs, e.g.:

```

File: docs/resources/rds_pg_plugin.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* `instance_id` - (Required, String, ForceNew) PostgreSQL instance ID. Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Plugin name. Changing this parameter will create a new resource.
+
+* `database_name` - (Required, String, ForceNew) Database name. Changing this parameter will create a new resource.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
```
Comment (houpeng80) [inline docs/resources/rds_pg_plugin.md:41]: id is missing

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/pagination"
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go:16]: add a ne line

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+	resp, err := pagination.ListAllItems(
+		client,
+		"offset",
+		reqPath,
+		&pagination.QueryOpts{MarkerField: ""})
+	if err != nil {
+		errCode := parseErrCode(resp)
+		if errCode == "DBS.280238" || errCode == "DBS.200823" {
+			return nil, golangsdk.ErrDefault404{}
+		}
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]:
```suggestion
		
		if errCode := parseErrCode(resp); errCode == "DBS.280238" || errCode == "DBS.200823" {
			return nil, golangsdk.ErrDefault404{}
		}
```

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+		return nil, fmt.Errorf("error marshaling RDS PG plugin: %s", err)
+	}
+
+	var bodyJson interface{}
+	err = json.Unmarshal(bodyBytes, &bodyJson)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshal PG plugin: %s", err)
+	}
+
+	log.Printf("[DEBUG] reading RDS PG plugin response: %#v", bodyJson)
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]:
```suggestion
	log.Printf("[DEBUG] Reading RDS PostgreSQL plugin response: %#v", bodyJson)
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getPgPluginResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.NewServiceClient("rds", acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating PG plugin Client: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating RDS Client: %s", err)
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	reqURI := "v3/{project_id}/instances/{instance_id}/extensions?database_name={database_name}"
+	reqPath := client.Endpoint + reqURI
+	reqPath = strings.ReplaceAll(reqPath, "{project_id}", client.ProjectID)
+	reqPath = strings.ReplaceAll(reqPath, "{instance_id}", state.Primary.Attributes["instance_id"])
+	reqPath = strings.ReplaceAll(reqPath, "{database_name}", state.Primary.Attributes["database_name"])
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go]:
```suggestion
	listPgPluginHttpUrl := "v3/{project_id}/instances/{instance_id}/extensions?database_name={database_name}"
	listPgPluginPath := client.Endpoint + reqURI
	listPgPluginPath = strings.ReplaceAll(reqPath, "{project_id}", client.ProjectID)
	listPgPluginPath = strings.ReplaceAll(reqPath, "{instance_id}", state.Primary.Attributes["instance_id"])
	listPgPluginPath = strings.ReplaceAll(reqPath, "{database_name}", state.Primary.Attributes["database_name"])
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+	resp, err := client.Request("GET", reqPath, &golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving PG plugin: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go]:
```suggestion
		return nil, fmt.Errorf("error retrieving RDS PostgreSQL plugin: %s", err)
```
and fix other places

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+	bodyParams := map[string]interface{}{
+		"database_name":  utils.ValueIngoreEmpty(d.Get("database_name")),
+		"extension_name": utils.ValueIngoreEmpty(d.Get("name")),
+	}
+	return bodyParams
+}
+
+func queryPluginDetail(client *golangsdk.ServiceClient, instanceId, databaseName, name string) (interface{}, error) {
+	reqURI := "v3/{project_id}/instances/{instance_id}/extensions?database_name={database_name}"
+	reqPath := client.Endpoint + reqURI
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]:
```suggestion
	listPgPluginHttpUrl := "v3/{project_id}/instances/{instance_id}/extensions?database_name={database_name}"
	listPgPluginPath := client.Endpoint + listPgPluginHttpUrl 
```
and fix other places

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildCreatePgPluginBody(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"database_name":  utils.ValueIngoreEmpty(d.Get("database_name")),
+		"extension_name": utils.ValueIngoreEmpty(d.Get("name")),
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]:
```suggestion
		"database_name":  d.Get("database_name"),
		"extension_name": d.Get("name"),
```

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+			return resource.RetryableError(err)
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return diag.Errorf("error creating RDS PG plugin: %s", err)
+	}
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]: fix it refer to the resource of huaweicloud_rds_pg_account

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+	region := cfg.GetRegion(d)
+	client, err := cfg.NewServiceClient("rds", region)
+	if err != nil {
+		return diag.Errorf("error creating RDS PG plugin client: %s", err)
+	}
+
+	arr := strings.Split(d.Id(), "/")
+	if len(arr) != 3 {
+		return diag.Errorf("invalid ID format in ReadContext: %s", d.Id())
+	}
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]:
```suggestion
	parts := strings.Split(d.Id(), "/")
	if len(parts) != 3 {
		return diag.Errorf("invalid ID format in ReadContext: %s", d.Id())
	}
```

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+		d.Set("description", utils.PathSearch("description", pluginDetail, "")),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildDeletePgPluginBody(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"database_name":  utils.ValueIngoreEmpty(d.Get("database_name")),
+		"extension_name": utils.ValueIngoreEmpty(d.Get("name")),
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]:
```suggestion
		"database_name": d.Get("database_name"),
		"extension_name": d.Get("name"),
```

File: huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+		if retryable {
+			return resource.RetryableError(err)
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error deleting RDS PG plugin")
```
Comment (houpeng80) [inline huaweicloud/services/rds/resource_huaweicloud_rds_pg_plugin.go]:
```suggestion
		return diag.Errorf("error deleting RDS PostgreSQL plugin: %s", err)
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+			200,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDS PostgreSQL : %s", err)
+	}
+
+	body, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving  RDS PostgreSQL : %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go]: please delete extra spaces.

```suggestion
		return nil, fmt.Errorf("error retrieving RDS PostgreSQL : %s", err)
```

File: huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	name := state.Primary.Attributes["name"]
+	plugin := utils.PathSearch(fmt.Sprintf("extensions[?name=='%s']|[?created]|[0]", name), body, nil)
+
+	if plugin == nil {
+		return nil, fmt.Errorf("error, no RDS PostgreSQL  matching %s was found", name)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go]: ditto, and update the error message

```suggestion
		return nil, fmt.Errorf("no RDS PostgreSQL plugin matching %s was found", name)
```

**Pull request merged**